### PR TITLE
feat: new ./bin/testplane

### PR DESCRIPTION
**Pull request by miripiruni**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -39,7 +39,8 @@
     "test"
   ],
   "bin": {
-    "hermione": "./bin/hermione"
+    "hermione": "./bin/hermione",
+    "testplane": "./bin/hermione"
```
Comment (KuznetsovRoman) [inline package.json:43]: You still have one binary file, so `npx testplane` will launch `./bin/hermione`, and `{cliName: 'hermione'}` would be passed to `run` method either way.

File: test/src/cli/index.js [hunk 2]
```diff
@@ -38,11 +38,27 @@ describe("cli", () => {
...
+        });
+
+        it("should show information about testplane by default", async () => {
+            const defaultResult = info.configOverriding();
+
+            assert.isTrue(defaultResult.includes("testplane"));
+            assert.isFalse(defaultResult.includes("hermione"));
+        });
+
+        it("should show information about hermione by default", async () => {
```
Comment (KuznetsovRoman) [inline test/src/cli/index.js:56]: > by default

Does not look like `default`, if you pass `{ cliName: "hermione" }`

File: src/cli/info.js [hunk 3]
```diff
@@ -1,20 +1,24 @@
 "use strict";
 
-exports.configOverriding = `  Overriding config
+exports.configOverriding = (opts = {}) => {
```
Comment (KuznetsovRoman) [inline src/cli/info.js:3]: You lost two extra spaces at the beginning.

Here we could do this:
```js
exports.configOverriding = ({ cliName = "testplane" } = {}) => `  Overriding config
    To override any config...
```

This way we won't need to have extra `return`, and spacing would be untouched

File: src/cli/index.js [hunk 4]
```diff
@@ -39,16 +39,16 @@ process.on("unhandledRejection", (reason, p) => {
...
+    const program = new Command(opts.cliName || "testplane");
 
     program.version(pkg.version).allowUnknownOption().option("-c, --config <path>", "path to configuration file");
 
     const configPath = preparseOption(program, "config");
     hermione = Hermione.create(configPath);
 
     program
-        .on("--help", () => logger.log(info.configOverriding))
+        .on("--help", () => logger.log(info.configOverriding(opts)))
```
Comment (KuznetsovRoman) [inline src/cli/index.js:51]: name `configOverriding` looks like it is a string.

If you change its type to `Function`, i suggest to rename it to something like `getConfigOverriding`, to emphasize it is a function

File: src/cli/index.js [hunk 5]
```diff
@@ -39,16 +39,16 @@ process.on("unhandledRejection", (reason, p) => {
     }
 });
 
-exports.run = () => {
-    const program = new Command();
+exports.run = (opts = {}) => {
+    const program = new Command(opts.cliName || "testplane");
```
Comment (KuznetsovRoman) [inline src/cli/index.js:43]: Why do we need this?

File: test/src/cli/index.js [hunk 6]
```diff
@@ -38,11 +38,27 @@ describe("cli", () => {
...
+            const defaultResult = info.configOverriding({ cliName: "hermione" });
```
Comment (KuznetsovRoman) [inline test/src/cli/index.js:57]: So it is not a `defaultResult` eigher

**Review by DudaGod — COMMENTED** [review]
Branch `rebranding` should be add to the protection rules. Moreover you should enable tests in this branch inside `.github/workflows/node.js.yml` (temporarily)

File: package.json (hunk 1 above)
Comment (miripiruni) [reply package.json:43]: Fixed: `"testplane": "./bin/tesplane"`.

File: src/cli/index.js (hunk 4 above)
Comment (miripiruni) [reply src/cli/index.js:51]: My changes is optimized for future situation: just get rid of old staff.

I'm looking forward to delete opts here and in similar places.

File: src/cli/info.js (hunk 3 above)
Comment (miripiruni) [reply src/cli/info.js:3]: > You lost two extra spaces at the beginning.

thank you!

Fixed.

File: test/src/cli/index.js (hunk 2 above)
Comment (miripiruni) [reply test/src/cli/index.js:56]: Fixed

**Comment by miripiruni** [comment]
@DudaGod setup CI for new branch — done. Thanks.

@KuznetsovRoman PR updated. "Resolve conversation" buttons is up to you.

File: package.json [hunk 7]
```diff
@@ -39,7 +39,8 @@
     "test"
   ],
   "bin": {
-    "hermione": "./bin/hermione"
+    "hermione": "./bin/hermione",
+    "testplane": "./bin/testplane"
```
Comment (KuznetsovRoman) [inline package.json:43]: `./bin/testplane` is missing

File: .github/workflows/node.js.yml [hunk 8]
```diff
@@ -5,9 +5,9 @@ name: Node.js CI
 
 on:
     push:
-        branches: [master]
+        branches: [master, rebranding]
     pull_request:
-        branches: [master]
+        branches: [master, rebranding]
```
Comment (KuznetsovRoman) [inline .github/workflows/node.js.yml:10]: Your branch is called `gemini-testing:rebranding`, so the auto-checks are still not launching

File: src/cli/index.js (hunk 5 above)
Comment (KuznetsovRoman) [reply src/cli/index.js:43]: And still, why do we need this? What for?

File: test/src/cli/index.js (hunk 6 above)
Comment (KuznetsovRoman) [reply test/src/cli/index.js:57]: name `defaultResult` is a lie.

**Review by KuznetsovRoman — CHANGES_REQUESTED** [review]

File: src/cli/index.js (hunk 5 above)
Comment (miripiruni) [reply src/cli/index.js:43]: [src/cli/index.js](https://github.com/gemini-testing/hermione/pull/892/commits/d062638402a43920027c0b0c89eeb81fe9584167#diff-f00e7c78fa83a1316c2844a481cc2eeda7c7c2c4f958d39e9e457ddbc6b5c20f) used twice:

In bin/hermione and bin/testplane.

In future we will delete ./bin/hermione and opts in src/cli/index.js::run() also.

Right now, I introduced opts to pass cliName to compose proper help in CLI.

I decided to use opts as object with cliName field to make in more readable in on the call side.

Compate variants:

```
.run('hermione');
.run({ cliName: 'hermione' });
```

…for me second is much better.

Comment (KuznetsovRoman) [reply src/cli/index.js:43]: I mean, why do we need to pass `"hermione" / "testplane"` into Command constructor?

**Pull request merged**